### ISSUE TITLE
Not calling reducer with failed promise

### ISSUE THREAD

**Issue by eventhough** [acdlite/redux-actions#42]
I'm using redux-actions with redux-promise. When my promise succeeds, it hits my `handleActions` reducer as expected, but when it fails, nothing happens. The action is fired off but my handler never sees the failed FSA.

From devtools I see that the FSA has `error` property set to `true` as expected, but my reducer below does not get called.

```javascript
const reducers = handleActions({
  [FOO_ACTION]: (state, action) => {
    console.log(action); // when promise fails, I don't see this!
    return state;
  },
}, initialState);
```

Then I tried something like this but it didn't work. I'm pretty sure this is *not* how we're supposed to use `handleActions`.

```javascript
const reducers = handleActions({
  [FOO_ACTION]: handleAction(FOO_ACTION, {
    next: (state, action) => {
      console.log('woo hoo:', action);
      return state;
    },
    throw: (state, action) => {
      console.log('aww man:', action);
      return state;
    },
  }),
}, initialState);
```

Could someone shed some more light on how to handle failed promises?

**Comment by eventhough** [comment]
Okay I think I figured it out. Here is how I set up my reducer to handle both successful and failed promises:

```javascript
const reducers = handleActions({
  [FOO_ACTION]: {
    next: (state, action) => {
      console.log('success:', action);
      return state;
    },
    throw: (state, action) => {
      console.log('fail:', action);
      return state;
    },
  },
}, initialState);
```

Doing this feels a bit counterintuitive to me since I thought the idea behind "FSA" was to do away with things like `LOAD_SUCCESS` and `LOAD_FAILURE`. If we are treating payload errors as a first class concept, doesn't creating a `next` and `throw` handler defeat the purpose? I feel like I am basically doing the same thing, just in a different way. Thanks!

**Comment by lukewestby** [comment]
Your first example where you were not seeing the single reducer called in the error case is a bug that was fixed in #40 and should be included in the next release. Once it is published you'll be able to implement your reducer the way you wanted in your first example. Keep an eye out for it! In the mean time I'm going to close this, but feel free to continue commenting if you have any additional questions or concerns.

**Comment by eventhough** [comment]
Awesome! Thanks Luke.
